new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java

```diff
@@ -605,6 +612,172 @@ public static HelmParams installAndVerifyVoyager(String voyagerNamespace,
+        null, null, null);
+  }
+
+  /**
+   * Install Apache and wait up to five minutes until the Apache pod is ready.
+   *
+   * @param apacheNamespace the namespace in which the Apache will be installed
+   * @param image the image name of Apache webtier
+   * @param httpNodePort the http nodeport of Apache
+   * @param httpsNodePort the https nodeport of Apache
+   * @param domainUid the uid of the domain to which Apache will route the services
+   * @param volumePath the path to put your own custom_mod_wl_apache.conf file
+   * @param virtualHostName the VirtualHostName of the Apache HTTP server which is used to enable custom SSL config
+   * @param clusterNamePortMap the map with clusterName as key and cluster port number as value
+   * @return the Apache Helm installation parameters
+   */
+  public static HelmParams installAndVerifyApache(String apacheNamespace,
+                                                  String image,
+                                                  int httpNodePort,
+                                                  int httpsNodePort,
+                                                  String domainUid,
+                                                  String volumePath,
+                                                  String virtualHostName,
+                                                  LinkedHashMap<String, String> clusterNamePortMap)
+      throws IOException {
+
+    LoggingFacade logger = getLogger();
+
+    // Create Docker registry secret in the apache namespace to pull the Apache webtier image from repository
+    if (!secretExists(REPO_SECRET_NAME, apacheNamespace)) {
+      logger.info("Creating Docker registry secret in namespace {0}", apacheNamespace);
+      createDockerRegistrySecret(apacheNamespace);
+    }
+
+    // map with secret
+    Map<String, Object> secretNameMap = new HashMap<>();
+    secretNameMap.put("name", REPO_SECRET_NAME);
+
+    // Helm install parameters
+    HelmParams apacheHelmParams = new HelmParams()
+        .releaseName(APACHE_RELEASE_NAME + "-" + apacheNamespace.substring(3))
+        .namespace(apacheNamespace)
+        .chartDir(APACHE_SAMPLE_CHART_DIR);
+
+    // Apache chart values to override
+    ApacheParams apacheParams = new ApacheParams()
+        .helmParams(apacheHelmParams)
+        .imagePullSecrets(secretNameMap)
+        .image(image)
+        .imagePullPolicy("Always")
+        .domainUID(domainUid);
+
+    if (httpNodePort >= 0 && httpsNodePort >= 0) {
+      apacheParams
+          .httpNodePort(httpNodePort)
+          .httpsNodePort(httpsNodePort);
+    }
+
+    if (volumePath != null && clusterNamePortMap != null) {
+      // create a custom Apache plugin configuration file named custom_mod_wl_apache.conf
+      Path customConf = Paths.get(volumePath, "custom_mod_wl_apache.conf");
+      ArrayList<String> lines = new ArrayList<>();
+      lines.add("<IfModule mod_weblogic.c>");
+      lines.add("WebLogicHost ${WEBLOGIC_HOST}");
```

Comment:
are these values going to be replaced?

Reply:
add more comments in this method to explain how it works

Reply:
this is not going to be replaced.
Also added more comments in the code.